.github/workflows/ci.yml

```diff
@@ -112,16 +122,30 @@ jobs:
     # This prevents duplicate CI runs for our own pull requests, whilst preserving the ability to
     # run the CI for each branch push to a fork, and for each pull request originating from a fork.
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
-    runs-on: windows-latest
+    needs: setup-os-matrix
+    strategy:
+      matrix:
+        os: ${{ fromJson(needs.setup-os-matrix.outputs.os) }}
+        library: [ILGPU, ILGPU.Algorithms]
+        framework: [netcoreapp3.1, net5.0]
+        include:
+          - os: windows-latest
+            library: [ILGPU, ILGPU.Algorithms]
+            framework: net47
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Setup .NET Core SDK
+
+      # .NET 5.0.x
+      - name: Setup .NET SDK v5.0.x
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.407
+          dotnet-version: 5.0.x
       - name: Pin .NET Core SDK
-        run: dotnet new globaljson --sdk-version 3.1.407
+        run: dotnet new globaljson --sdk-version 5.0.x
+
```

Comment:
same remarks as above about action name and no need for pinning

Reply:
resolved with previus 🙏